qemu/tests/ovs_host_vlan.py

```diff
@@ -0,0 +1,293 @@
+        logging.debug("%s" % output)
+    elif os_type == "windows":
+        ping_cmd = "ping %s -n %s" % (dest, count)
+        status, output = session.cmd_status_output(ping_cmd, timeout=60)
+        logging.debug("output: %s" % output)
+        if match_error in str(output):
+            ratio = 100
+        else:
+            ratio = utils_test.get_loss_ratio(output)
+        ping_result_check(test, ratio, same_vlan)
+
+
+def netperf_setup(test, params, env):
+    """
+    Setup netperf in guest.
+
+    Copy netperf package into guest. Install netperf in guest (linux only).
+    """
+    tmp_params = params
+    tmp_params["start_vm"] = "yes"
+    tmp_params["image_snapshot"] = "no"
+    vm_name = params.get("main_vm")
+    env_process.preprocess_vm(test, tmp_params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
+    try:
+        if params.get("os_type") == "linux":
+            netperf_link = params["netperf_link"]
+            netperf_path = params["netperf_path"]
+            src_link = os.path.join(data_dir.get_deps_dir("netperf"), netperf_link)
+            vm.copy_files_to(src_link, netperf_path, timeout=60)
+            setup_cmd = params.get("setup_cmd")
+            (status, output) = session.cmd_status_output(setup_cmd % netperf_path,
+                                                         timeout=600)
+            if status != 0:
+                err = "Fail to setup netperf on guest os."
+                err += " Command output:\n%s" % output
+                test.error(err)
+        elif params.get("os_type") == "windows":
+            s_link = params.get("netperf_server_link_win", "netserver-2.6.0.exe")
+            src_link = os.path.join(data_dir.get_deps_dir("netperf"), s_link)
+            netperf_path = params["netperf_path"]
+            vm.copy_files_to(src_link, netperf_path, timeout=60)
+            s_link = params.get("netperf_client_link_win", "netperf.exe")
+            src_link = os.path.join(data_dir.get_deps_dir("netperf"), s_link)
+            vm.copy_files_to(src_link, netperf_path, timeout=60)
+    finally:
+        if session:
+            session.close()
+        vm.destroy()
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    QEMU 'open vswitch host vlan' test
+
+    1) Start a VM and setup netperf in it.
+    2) Stop NetworkManager service in host.
+    3) Create a new private ovs bridge, which has no physical nics inside.
+    4) Create 4 ovs ports and add to 2 vlans.
+    4) Boot 4 VMs on this bridge and add them to 2 vlans.
+    5) Configure ip address of all systems make sure all IPs are in same subnet.
+    6) Ping between two guests in same vlan.
+    7) Ping between two guests in different vlan.
+    8) Ping between two guests in another vlan.
+    9) Netperf test between two guests in same vlan.
+    10) Transfer file between to guests in same vlan. (optional)
```

Comment:
Typo, and you missed steps in `finally:`

Reply:
could you help me out of what did i miss?

Reply:
ovs port remove on host side

Reply:
if the bridge is removed, all the port attached to this bridge will be removed.